client/src/reducers/datasetMetadata.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ Dataset metadata reducer, modifies Portal collections-related state. 
+ */
+
+// Core dependencies
+import { Action } from "redux";
+
+// App dependencies
+import { DatasetMetadata as IDatasetMetadata } from "../common/types/entities";
+
+/*
+ Action dispatched on successful response from dataset-metadata endpoint.
+ */
+export interface DatasetMetdataAction extends Action<string> {
+  datasetMetadata: IDatasetMetadata;
+  error: string;
+  portalUrl: string;
+}
+
+/*
+ Dataset metdata state; selected dataset ID and corresponding collection information.
+ */
+export interface DatasetMetadataState {
+  datasetMetadata: IDatasetMetadata | null;
+  error: string | null;
+  loading: boolean;
+  portalUrl: string | null;
+}
+
+const DatasetMetadata = (
+  state: DatasetMetadataState = {
+    // data loading flag
+    loading: true,
+    error: null,
+
+    datasetMetadata: null,
+    portalUrl: null,
+  },
+  action: DatasetMetdataAction
+): DatasetMetadataState => {
+  switch (action.type) {
+    case "initial data load start":
```

Comment:
Using descriptive types with spaces seems a bit odd to me. @tihuan what is the standard in this project?

Reply:
Ideally I think using ENUMS to denote all the action types would prevent typos and duplicates, but that's out of scope of this PR, so I think what it is now is fine 😆